src/main/java/com/autonomic/tmc/auth/exception/BaseSdkException.java

```diff
@@ -19,75 +19,88 @@
  */
 package com.autonomic.tmc.auth.exception;
 
-import java.io.IOException;
-import java.net.URISyntaxException;
-import java.util.Objects;
-import java.util.Optional;
+import static java.util.Optional.ofNullable;
+
 import java.util.jar.JarFile;
 import java.util.jar.Manifest;
 import lombok.extern.slf4j.Slf4j;
 
 @Slf4j
 public class BaseSdkException extends RuntimeException {
 
-    private static final String PROJECT_NAME;
-    private static final String PROJECT_VERSION;
-
-    static {
-        ProjectProperties projectProperties = new ProjectProperties();
-        PROJECT_NAME = projectProperties.getName();
-        PROJECT_VERSION = projectProperties.getVersion();
-    }
+    private static final String DEFAULT_NAME = "[ AUTONOMIC ]";
+    private static final String DEFAULT_VERSION = "[ SDK ]";
 
     BaseSdkException(String message) {
         super(message);
+        log.warn(message);
     }
 
     BaseSdkException(String message, Throwable cause) {
         super(message, cause);
+        log.warn(message, cause);
     }
 
     static String buildMessage(ErrorSourceType errorSourceType, String clientMessage) {
-        return String
-            .format("%s-%s-%s: %s.", PROJECT_NAME, PROJECT_VERSION, errorSourceType, clientMessage);
+        try {
+            final ProjectProperties properties = ProjectProperties.get();
+            final String name = ofNullable(properties.getName()).orElse(DEFAULT_NAME);
+            final String version = ofNullable(properties.getVersion()).orElse(DEFAULT_VERSION);
+            return String.format("%s-%s-%s: %s.", name, version, errorSourceType, clientMessage);
+        } catch (Throwable e) {
+            return DEFAULT_NAME + "~" + DEFAULT_VERSION + "~" + errorSourceType.toString() + clientMessage;
```

Comment:
`String.format` could throw an exception 🤣 

Therefore, we catch and try another way in the event that happens...
Technically, its not exactly the same.... we made it, intentionally subtle, the difference between the 2 lines so that we will be aware when this happens in the future so we can better address it.

We took this story to the levels of paranoia to ensure no exceptions could possibly occur.